MBBSEmu/Logging/Targets/QueueTarget.cs

```diff
@@ -23,13 +23,22 @@ public QueueTarget(int maxQueueSize = 1000)
         /// <summary>
         ///     ILoggingTarget Implementation
         /// </summary>
-        /// <param name="logEntry"></param>
-        /// <exception cref="NotImplementedException"></exception>
+        /// <param name="logEntry">params object[] consisting of (logLevel, logMessage)</param>
         public void Write(params object[] logEntry)
         {
+            //Use reflection to get the name of the class calling this method
+            //TODO: We should move this reflection up the stack to the log factory in GetLogger<T>() and pass it into the logger so we don't have to do this every time
+            var callingType = new System.Diagnostics.StackTrace().GetFrame(3)?.GetMethod()?.DeclaringType;
```

Comment:
at least in Java fetching the stack trace is an expensive operation. Given the level of logging we do this could slow things down?

Reply:
https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.callermembernameattribute?view=net-7.0

https://stackoverflow.com/questions/1348643/how-performant-is-stackframe

Reply:
The issue here is that the "Calling Member" of the Target is the Logger. Because of this, we have to get the current stack and look up 3 frames to get the actual logging class. I'm not _entirely_ concerned about the performance atm as in Release mode, the default logging is trimmed down to be pretty minimal and I'd suspect not a critical performance issue.

Again, something to keep in mind for future refactor and moving this up to the Factory level when `GetLogger<T>` is called.

Reply:
Potential Work Around: https://www.c-sharpcorner.com/article/boosting-up-the-reflection-performance-in-c-sharp/